environment_switcher/lib/src/environment_switcher.dart

```diff
@@ -0,0 +1,195 @@
+import 'package:flutter/material.dart';
+
+import 'environment.dart';
+import 'environment_store.dart';
+import 'select_environment_sheet.dart';
+
+export 'environment.dart';
+export 'environment_store.dart';
+
+class EnvironmentSwitcher extends StatefulWidget {
+  final Widget Function(Environment) builder;
+  final List<Environment> environments;
+  final Environment defaultEnvironment;
+  final EnvironmentStore environmentStore;
+  final bool showBanner;
+
+  /// A banner that visually shows the user what [Environment]
+  /// is currently set to.
+  /// This is useful because it's possible to be on one Flavour
+  /// in the native code, and another in the Flutter code.
+  const EnvironmentSwitcher({
+    @required this.builder,
+    @required this.environments,
+    this.environmentStore, // mainly required for testing
+    this.showBanner = true,
+    this.defaultEnvironment,
+  })  : assert(builder != null),
+        assert(environments != null && environments.length != 0),
+        assert(showBanner != null),
+        assert(showBanner ? defaultEnvironment != null : true);
+
+  @override
+  State<StatefulWidget> createState() {
+    return _StateEnvironmentSwitcher(environments, environmentStore);
+  }
+}
+
+class _StateEnvironmentSwitcher extends State<EnvironmentSwitcher> {
+  final List<Environment> environments;
+  final EnvironmentStore environmentStore;
+
+  Environment currentEnvironment;
```

Comment:
Is there any way for a child widget to get the current environment. If not, maybe use an `InheritedWidget` now that we can't use `BlocProvider`